examples/visionProcessing/genderDetection/genderDetection.cpp

```diff
@@ -0,0 +1,207 @@
+      measurementHandler->setField(std::to_string(computeCounter), "CMP_START",
+                                   0);
+      measurementHandler->setField(std::to_string(frameID), "FD->GD", 0);
+
+      cv::Mat blob = cv::dnn::blobFromImage(greyImage, 1, cv::Size(227, 227),
+                                            MODEL_MEAN_VALUES, false);
+
+      genderNet.setInput(blob);
+      cv::Mat genderPreds = genderNet.forward();
+
+      // Get gender prediction (assumed binary classification, 0 = Male, 1 =
+      // Female)
+      std::vector<float> genderPredsVec;
+      genderPreds.reshape(1, 1).copyTo(genderPredsVec);
+      std::string gender = (genderPredsVec[0] > 0.5) ? "Female" : "Male";
+
+      // Serialize results
+      nlohmann::json resultData;
+      resultData["frameID"] = frameID;
+      resultData["Gender"] = gender;
+
+      std::vector<uint8_t> encodedAnalytics = Serde::serialize(resultData);
+
+      std::cout << "GenderDetection: \n"
+                << " FrameID:" << frameID << "\n"
+                << " Gender: " << gender << "\n"
+                << " Encoded Analytics Size: " << encodedAnalytics.size()
+                << " bytes" << std::endl;
+      std::cout << "------------------------------------" << std::endl;
+
+      push(encodedAnalytics);
+      measurementHandler->setField(std::to_string(frameID), "GD->AGG", 0);
+      measurementHandler->setField(std::to_string(computeCounter), "CMP_FINISH",
+                                   0);
+      computeCounter++;
+    }
+  }
+
+private:
+  cv::dnn::Net genderNet;
+  cv::Scalar MODEL_MEAN_VALUES =
+      cv::Scalar(78.4263377603, 87.7689143744, 114.895847746);
+  int computeCounter = 0;
+};
+
+std::vector<uint32_t> createConsumerPartitions(uint32_t highestPartitionNumber,
+                                               uint32_t consumerIndex,
+                                               uint32_t numberOfConsumers) {
+  std::vector<uint32_t> consumerPartitions;
+  for (auto i = consumerIndex; i <= highestPartitionNumber;
+       i += numberOfConsumers) {
+    consumerPartitions.push_back(i);
+  }
+  return consumerPartitions;
+}
+
+void run(const std::string &nodeName, const std::string &dagFileName,
+         uint32_t consumerIndex, uint32_t numberOfConsumers,
+         const std::string &protobufFile, const std::string &mlModel) {
+
+  GenderDetection genderDetection(protobufFile, mlModel);
+  ndn::Face face;
+
+  auto dagParser = iceflow::DAGParser::parseFromFile(dagFileName);
+  auto node = dagParser.findNodeByName(nodeName);
+  if (!node.downstream.empty()) {
+    std::string downstreamEdgeName;
+    for (const auto &downstream : node.downstream) {
+      if (downstream.id == "gd2agg") {
+        downstreamEdgeName = downstream.id;
+      }
+    }
+
+    if (downstreamEdgeName.empty()) {
+      std::cerr << "Error: Missing downstream target for gender detection."
+                << std::endl;
+      return;
+    }
```

Comment:
Same here, I think this might be a bit too complicated and should not actually work when you try to run it.